### PR TITLE
[air/output] Improve leaked mentions of Tune concepts

### LINKED ISSUE #33839: [AIR output] Tune concepts are leaked when I only used a trainer
### What happened + What you expected to happen

1. Ran pytorch_training_e2e.py. The warning is very confusing since it mentions `trials` and `tune`
```
(autoscaler +10s) Adding 1 node(s) of type worker-node-type-1.
2023-03-28 15:27:35,585 WARNING insufficient_resources_manager.py:128 -- Ignore this message if the cluster is autoscaling. You asked for 1.0 cpu and 1.0 gpu per trial, but the cluster only has 0 cpu and 0 gpu. Stop the tuning job and adjust the resources requested per trial (possibly via `resources_per_trial` or via `num_workers` for rllib) and/or add more resources to your Ray runtime.
```

2. Ran` tensorflow_benchmark.py` in release test with `AIR_VERBOSITY=1`

When trying to force stop the execution via `cmd+c`. I saw messages related to `trial` and `experiment`
```
^C2023-03-28 18:19:28,738       WARNING tune.py:185 -- Stop signal received (e.g. via SIGINT/Ctrl+C), ending Ray Tune run. This will try to checkpoint the experiment state one last time. Press CTRL+C (or send SIGINT/SIGKILL/SIGTERM) to skip. 
```
and
```
2023-03-28 18:19:09,796 ERROR tune.py:941 -- Trials did not complete: [TensorflowTrainer_886ed_00000]
2023-03-28 18:19:09,797 WARNING tune.py:955 -- Experiment has been interrupted, but the most recent state was saved.
Continue running this experiment with: Tuner.restore(path="/home/ray/ray_results/TensorflowTrainer_2023-03-28_18-17-54", trainable=...)
```

3. When I ran `air_benchmark_xgboost_cpu_10 `
```
2023-03-28 13:51:14,756 WARNING trial_runner.py:1576 -- The maximum number of pending trials has been automatically set to the number of available cluster CPUs, which is high (176 CPUs/pending trials). If you're running an experiment with a large number of trials, this could lead to scheduling overhead. In this case, consider setting the `TUNE_MAX_PENDING_TRIALS_PG` environment variable to the desired maximum number of concurrent trials.
```

```
2023-03-28 14:04:06,409 WARNING util.py:244 -- The `process_trial_save` operation took 3.927 s, which may be a performance bottleneck.
2023-03-28 14:04:06,409 WARNING trial_runner.py:887 -- Consider turning off forced head-worker trial checkpoint syncs by setting sync_on_checkpoint=False. Note that this may result in faulty trial restoration if a failure occurs while the checkpoint is being synced from the worker to the head node.
```

### Versions / Dependencies

nightly

### Reproduction script

workloads/tensorflow_benchmark.py in release test

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

Ray Tune is the execution backend for Ray Train. This means that sometimes error/warning messages use Tune concepts, that don't make sense in a single-trial run, such as with Ray Train trainers.

This PR improves three such occurrences:

1. The insufficient resources warnings message has been adjusted in the case where only one trial is run
2. Calculation of `max_pending_trials` now uses `search_alg.total_samples` as the minimum, which was an oversight before.
3. On interrupt of a training run, a `Tuner.restore()` message was suggested, but it should be `Trainer.restore()`

With these fixes, we will see message such as:

**Insufficient resources**

```
Training has not started in the last 30 seconds.
This could be due to the cluster not having enough resources available.
You asked for 4 CPUs and 2 GPUs, but the cluster only 
has 4 CPUs and 1 GPUs available.
Stop the training and adjust the required resources (e.g. via the 
`ScalingConfig` or `resources_per_trial`, or `num_workers` for rllib), 
or add more resources to your cluster.
```

**Maximum number of pending trials**

The message

```
2023-03-28 13:51:14,756 WARNING trial_runner.py:1576 -- The maximum number of pending trials has been automatically set to the number of available cluster CPUs, which is high (176 CPUs/pending trials). If you're running an experiment with a large number of trials, this could lead to scheduling overhead. In this case, consider setting the `TUNE_MAX_PENDING_TRIALS_PG` environment variable to the desired maximum number of concurrent trials.
```

will not turn up for Ray Train runs anymore.

**Restore**

For Train runs, will now be:

```
Experiment has been interrupted, but the most recent state was saved.
Continue running this experiment with: Trainer.restore(path="/Users/kai/ray_results/debug_execution_restart_dpt7", ...)
```

We can't get rid of all tune-related concepts just yet, but this is at least an improvement.

## Related issue number

Closes #33839 (or parts of it)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — CHANGES_REQUESTED** [review]
I like the error message refactoring.
want to discuss about the long term plan of how to differentiate between trainer and tuner though.

File: python/ray/train/base_trainer.py [hunk 1]
```diff
@@ -578,7 +578,10 @@ def fit(self) -> Result:
             )
         else:
             tuner = Tuner(
-                trainable=trainable, param_space=param_space, run_config=self.run_config
+                trainable=trainable,
+                param_space=param_space,
+                run_config=self.run_config,
+                _trainer_api=True,
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/train/base_trainer.py:584]: I think we generally need a good idea how to pass this information.

To me it feels like there should be some time of context. We have different requirements for different ML jobs. Even rllib vs Train has different requirements (e.g. default metrics to show), and maybe even rllib's single algorithms.

We don't have that story, yet, so in order to unblock this work, I think we can go ahead with the private flags. But yes, we should resolve this (also for telemetry).

File: python/ray/tune/execution/trial_runner.py [hunk 2]
```diff
@@ -133,12 +133,15 @@ def __init__(
         callbacks: Optional[List[Callback]] = None,
         metric: Optional[str] = None,
         trial_checkpoint_config: Optional[CheckpointConfig] = None,
+        _trainer_api: bool = False,
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/execution/trial_runner.py:136]: I've tried this before, and long story short, it's not very straightforward due to the fact that we need some of the information pretty early, but number of trials is only calculated later. It also can lead to confusing situations - e.g. it's totally valid to use `Tuner(trainable, tune_config=TuneConfig(num_samples=1))` for iteration and still expect to see a `Tuner.restore()` hint at the end. Tracking which object actually was the entrypoint saves us from those problems.

**Comment by krfricke** [comment]
Updated, thanks!

File: python/ray/tune/execution/trial_runner.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gjoliver) [reply python/ray/tune/execution/trial_runner.py:137]: ok, I just feel like passing such a generic sounding parameter all the way through so many components like Tuner, BackendExecutor, etc  just to be able to show the right output message is kind of too heavy.
we should probably have a way out of this if we want to live with it for now. @justinvyu 
another idea is maybe passing this bit through a LoggingConfig or something? we probably need such a config class anyways?

[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/execution/trial_runner.py:137]: We're doing the same at the moment with `_tuner_api`, so in that sense it's consistent :-D 

This is not a logging _configuration_ in my opinion. Users should not "configure" which output/error messages they want to see. It's more of a runtime context.

Ray core has a runtime context object, I think we need something similar for Ray AIR. Maybe we can even extend the Ray runtime context with a field for application-specific information.

**Review by gjoliver — APPROVED** [review]
sorry I missed this.
let's add a TODO and unblock the output work.

**Pull request merged**